server/models/procedures/trial_balance.sql

```diff
@@ -96,91 +96,107 @@ CREATE PROCEDURE TrialBalanceErrors()
 BEGIN
 
   -- this will hold our error cases
-  CREATE TEMPORARY TABLE IF NOT EXISTS stage_trial_balance_errors (record_uuid BINARY(16), trans_id TEXT, code TEXT);
+  CREATE TEMPORARY TABLE IF NOT EXISTS stage_trial_balance_errors (
+    record_uuid BINARY(16),
+    trans_id TEXT,
+    account_id int(10) unsigned,
+    account_label TEXT,
+    code TEXT
+  );
 
   -- check if dates are in the correct period
   INSERT INTO stage_trial_balance_errors
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.DATE_IN_WRONG_PERIOD' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, CONCAT(pj.trans_id, ' : ', pj.trans_date)  as error_description, 'POSTING_JOURNAL.ERRORS.DATE_IN_WRONG_PERIOD' AS code
     FROM posting_journal AS pj
       JOIN stage_trial_balance_transaction AS temp ON pj.record_uuid = temp.record_uuid
       JOIN period AS p ON pj.period_id = p.id
+      JOIN account AS ac ON ac.id = pj.account_id
     WHERE DATE(pj.trans_date) NOT BETWEEN DATE(p.start_date) AND DATE(p.end_date)
     GROUP BY pj.record_uuid;
 
   -- check to make sure that the fiscal year is not closed
   INSERT INTO stage_trial_balance_errors
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.CLOSED_FISCAL_YEAR' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, fiscal_year.label  as error_description,'POSTING_JOURNAL.ERRORS.CLOSED_FISCAL_YEAR' AS code
     FROM posting_journal AS pj JOIN stage_trial_balance_transaction AS temp
       ON pj.record_uuid = temp.record_uuid
       JOIN fiscal_year ON pj.fiscal_year_id = fiscal_year.id
+      JOIN account AS ac ON ac.id = pj.account_id
     WHERE fiscal_year.locked <> 0
     GROUP BY pj.record_uuid;
 
   -- check to make sure that all lines of a transaction have a description
   INSERT INTO stage_trial_balance_errors
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.MISSING_DESCRIPTION' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, pj.trans_id  as error_description, 'POSTING_JOURNAL.ERRORS.MISSING_DESCRIPTION' AS code
     FROM posting_journal AS pj JOIN stage_trial_balance_transaction AS temp
       ON pj.record_uuid = temp.record_uuid
+      JOIN account AS ac ON ac.id = pj.account_id
     WHERE pj.description IS NULL
     GROUP BY pj.record_uuid;
 
   -- check that all periods are unlocked
   INSERT INTO stage_trial_balance_errors
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.LOCKED_PERIOD' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, CONCAT(ac.number,' - ', ac.label)  as error_description, 'POSTING_JOURNAL.ERRORS.LOCKED_PERIOD' AS code
```

Comment:
Let's just remove `ac.id` from all these.  It looks like we don't use it on the client anyway..